src/main/java/org/opentripplanner/transit/model/network/SubMode.java

```diff
@@ -0,0 +1,122 @@
+package org.opentripplanner.transit.model.network;
+
+import java.io.Serializable;
+import java.util.BitSet;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class wrap a string and create an index for it, so we can use BitSet for matching a
+ * trip netexSubMode. SubModes are used in trip filtering for every request.
+ * <p>
+ * Naming, this class is named SubMode, not NetexSubMode because we want to migrate gtfsRouteType
+ * into the same concept.
+ * <p>
+ * This class is thread-safe, and the performance overhead should affect the graph built time,
+ * but not routing.
```

Comment:
```suggestion
 * This class wraps a string and creates an index for it, so we can use a BitSet for matching a
 * trip netexSubMode. SubModes are used in trip filtering for every request.
 * <p>
 * Naming; this class is named SubMode, not NetexSubMode because we want to migrate gtfsRouteType
 * into the same concept.
 * <p>
 * This class is thread-safe, and the performance overhead should affect the graph built time,
 * but not routing.
```